dot/rpc/modules/api.go

```diff
@@ -93,3 +95,12 @@ type SystemAPI interface {
 	ChainType() string
 	ChainName() string
 }
+
+// BlockFinalityAPI is the interface for handling block finalisation methods
+type BlockFinalityAPI interface {
+	GetSetID() uint64
+	GetRound() uint64
+	GetVoters() grandpa.Voters
+	PreVotes() (map[ed25519.PublicKeyBytes]*grandpa.Vote, map[ed25519.PublicKeyBytes][]*grandpa.Vote)
+	PreCommits() (map[ed25519.PublicKeyBytes]*grandpa.Vote, map[ed25519.PublicKeyBytes][]*grandpa.Vote)
+}
```

Comment:
since the `*grandpa.Vote`s aren't actually needed by these rpc calls, it would be nicer for these functions to just return `[]ed25519.PublicKeyBytes` of all the authorities that have voted. that would simplify the logic in the `RoundState` function as well

Reply:
updated